hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/TimesTenDialect.java

```diff
@@ -157,21 +177,133 @@ public int getDefaultDecimalPrecision() {
 	public void initializeFunctionRegistry(FunctionContributions functionContributions) {
 		super.initializeFunctionRegistry(functionContributions);
 
-		CommonFunctionFactory functionFactory = new CommonFunctionFactory(functionContributions);
-		functionFactory.trim2();
-		functionFactory.soundex();
-		functionFactory.trunc();
-		functionFactory.toCharNumberDateTimestamp();
-		functionFactory.ceiling_ceil();
+		final TypeConfiguration typeConfiguration = functionContributions.getTypeConfiguration();
+		CommonFunctionFactory functionFactory     = new CommonFunctionFactory(functionContributions);
+		final BasicTypeRegistry basicTypeRegistry = typeConfiguration.getBasicTypeRegistry();
+		final BasicType<Date>   timestampType     = basicTypeRegistry.resolve( StandardBasicTypes.TIMESTAMP );
+		final BasicType<Date>   dateType          = basicTypeRegistry.resolve( StandardBasicTypes.DATE );
+		final BasicType<Date>   timeType          = basicTypeRegistry.resolve( StandardBasicTypes.TIME );
+		final BasicType<String> stringType        = basicTypeRegistry.resolve( StandardBasicTypes.STRING );
+		final BasicType<Long>   longType          = basicTypeRegistry.resolve( StandardBasicTypes.LONG );
+		final BasicType<Integer>intType           = basicTypeRegistry.resolve( StandardBasicTypes.INTEGER );
+
+		// String Functions
+		functionContributions.getFunctionRegistry().register( 
+				"rtrim", new StandardSQLFunction("rtrim", StandardBasicTypes.STRING) 
+		);
+		functionContributions.getFunctionRegistry().register( 
+				"ltrim", new StandardSQLFunction("ltrim", StandardBasicTypes.STRING) 
+		);
```

Comment:
Why not use the standard trim function definitions?

```suggestion
		functionFactory.trim2();
```